rhsm-client/src/main/java/org/candlepin/subscriptions/conduit/rhsm/client/X509ApiClientFactoryConfiguration.java

```diff
@@ -110,7 +111,7 @@ public boolean usesClientAuth() {
     }
 
     public boolean usesDefaultTruststore() {
-        return getTruststoreFile() == null;
+        return !StringUtils.hasText(getTruststoreFile());
```

Comment:
Note: needed this to reuse the RHSM_KEYSTORE without specifying truststore.

Reply:
I'm not trying to combine the stores. Sorry for the confusion.

I wanted to prevent the proliferation of so many aliases for the same keystore, so I decided to re-use the same env var that is already used for the RHSM API client. However, currently there's some buggy logic: if you set `RHSM_KEYSTORE`, then `RhsmApiFactory` logic falls into the branch where it uses `X509ApiClientFactory`. `X509ApiClientFactory` currently checks `usesDefaultTruststore` and without this change, because we define truststore file as `truststore-file: ${RHSM_TRUSTSTORE:}`, it is never null. Then when the logic tries to load a truststore, it fails. This isn't an issue so far in practice because we always specify both together. However, this PR makes it valid to specify just the keystore (but not the truststore)... This change was simply to fix that bug.